07-http_gordle/final/internal/handlers/newgame/newgame.go

```diff
@@ -0,0 +1,70 @@
+package newgame
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+
+	"learngo-pockets/httpgordle/internal/domain"
+	"learngo-pockets/httpgordle/internal/gordle"
+	"learngo-pockets/httpgordle/internal/handlers/apiconversion"
+)
+
+type gameCreator interface {
+	Add(domain.Game) error
+}
+
+// Handler returns the handler for the game creation endpoint.
+func Handler(repo gameCreator) http.HandlerFunc {
+	return func(writer http.ResponseWriter, _ *http.Request) {
+		game, err := create(repo)
+		if err != nil {
+			log.Printf("unable to create a new game: %s", err)
```

Comment:
Avoid using the global logger as it mangles the log output in concurrent tests.

Reply:
Explained in the chapter.